src/client.rs

```diff
@@ -399,3 +414,100 @@ pub enum ListOffsetsPosition {
     /// Fetch the offset for the corresponding timestamp.
     Timestamp(i64),
 }
+
+pub struct Admin {
+    /// The client handle from which this producer was created.
+    _client: Client,
+    /// The channel used by this producer.
+    tx: mpsc::UnboundedSender<BrokerResponse>,
+    /// The channel used by this producer.
+    rx: mpsc::UnboundedReceiver<BrokerResponse>,
```

Comment:
In order to make the Admin client cheaply cloneable, maybe we should stick with using `oneshot` channels internally in each of these methods. 

I'm only advocating that because I would like to cut the system over to using the multishot crate with a pool of re-useable multishots. That will remove the allocations, while still providing a really light-weight interface. IE, can clone the admin interface and share it around as much as you want, as it would mostly just be a wrapper around the underlying client.

Reply:
Yeah, I think I did this before we adopted multishot. Will use it